synapse/storage/databases/main/transactions.py

```diff
@@ -145,7 +147,11 @@ async def set_received_txn_response(
             desc="set_received_txn_response",
         )
 
-    async def get_destination_retry_timings(self, destination):
+    @cached(max_entries=10000)
```

Comment:
10k destinations seems high, but maybe it isn't? (Did you just grab a random number or did you check the previous size of the cache?)

Reply:
The flip side was that 1000 felt a bit small. I mostly just pulled it out of thin air, and it matches some other "small" caches.

Reply:
Ah, yeah I guess 1000 doesn't feel big enough. 👍 Was mostly curious where it came from. Thin air seems OK to me. ✈️